### PR TITLE
Fix BN stats to not expand shape

### PR DESCRIPTION
**Pull request by awni**
## Proposed changes

This will break loading in strict mode which checks the weight shapes match.

Noticed in https://github.com/ml-explore/mlx-examples/pull/264

### CODE REVIEW

**Review by angeloskath — APPROVED** [review]
Fair enough. I guess the argument being that they can be applied to arbitrary batch sizes and/or PyTorch compatibility.

**Comment by awni** [comment]
Yes that's a nice feature. But actually the main reason for this change is simply doing `model.load_weights` would crash since the shape of the stats changes after the first application of `BatchNorm`. Another option would be to make `load_weights` less strict (like broadcast is ok)..but this seems a bit simpler.
